File: TypeScript/day03/part1.ts
import { getPuzzleInput, createAlphabet } from "../utils";
const input = getPuzzleInput("day03");

export default function part1(input: string) {
  const arr = input.split("\n");
  const letters = createAlphabet();
  let final = 0;

  arr.forEach((line) => {
    const c1 = line.slice(0, line.length / 2).split("");
    const c2 = line.slice(line.length / 2).split("");
    const common = new Set<string>();

    c1.forEach((v) => c2.includes(v) && common.add(v));
    common.forEach((l) => (final += letters.indexOf(l) + 1));
  });

  return final;
}

File: Go/day03/part1.go
package main

import (
	"adventofcode/packages/utils"
	"strings"
)

func part1() int {
	content := utils.GetPuzzleInput("day03")

	alphabet := utils.CreateAlphabet()
	p := 0

	for _, v := range content {
		c1 := v[0 : len(v)/2]
		c2 := v[len(v)/2:]
		common := make([]string, 0)

		for _, c := range c1 {
			if strings.Contains(c2, string(c)) {
				if !utils.Contains(common, string(c)) {
					common = append(common, string(c))
				}
			}
		}

		for _, v := range common {
			for i, l := range alphabet {
				if v == l {
					p += i + 1
				}
			}
		}
	}

	return p
}
